src/stcal/skymatch/skyimage.py

```diff
@@ -0,0 +1,1020 @@
+        self._sky = sky
+
+    @property
+    def is_sky_valid(self):
+        """
+        Indicates whether sky value was successfully computed.
+        Must be set externally.
+        """
+        return self._sky_is_valid
+
+    @is_sky_valid.setter
+    def is_sky_valid(self, valid):
+        self._sky_is_valid = valid
+
+    @property
+    def radec(self):
+        """
+        Get RA and DEC of the vertices of the bounding polygon as a
+        `~numpy.ndarray` of shape (N, 2) where N is the number of vertices + 1.
+        """
+        return self._radec
+
+    @property
+    def polygon(self):
+        """Get image's bounding polygon."""
+        return self._polygon
+
+    def intersection(self, skyimage):
+        """
+        Compute intersection of this `SkyImage` object and another
+        `SkyImage`, `SkyGroup`, or
+        :py:class:`~spherical_geometry.polygon.SphericalPolygon`
+        object.
+
+        Parameters
+        ----------
+        skyimage : SkyImage, SkyGroup, SphericalPolygon
+            Another object that should be intersected with this `SkyImage`.
+
+        Returns
+        -------
+        polygon : SphericalPolygon
+            A :py:class:`~spherical_geometry.polygon.SphericalPolygon` that is
+            the intersection of this `SkyImage` and `skyimage`.
+
+        """
+        if isinstance(skyimage, (SkyImage, SkyGroup)):
+            other = skyimage.polygon
+        else:
+            other = skyimage
+
+        pts1 = np.sort(list(self._polygon.points)[0], axis=0)
+        pts2 = np.sort(list(other.points)[0], axis=0)
+        if np.allclose(pts1, pts2, rtol=0, atol=5e-9):
+            intersect_poly = self._polygon.copy()
+        else:
+            intersect_poly = self._polygon.intersection(other)
+        return intersect_poly
+
+    def calc_bounding_polygon(self, stepsize=None):
+        """Compute image's bounding polygon.
+
+        Parameters
+        ----------
+        stepsize : int, None, optional
+            Indicates the maximum separation between two adjacent vertices
+            of the bounding polygon along each side of the image. Corners
+            of the image are included automatically. If `stepsize` is `None`,
+            bounding polygon will contain only vertices of the image.
+
+        """
+        ny, nx = self.image_shape
+
+        if stepsize is None:
+            nintx = 2
+            ninty = 2
+        else:
+            nintx = max(2, int(np.ceil((nx + 1.0) / stepsize)))
+            ninty = max(2, int(np.ceil((ny + 1.0) / stepsize)))
+
+        xs = np.linspace(-0.5, nx - 0.5, nintx, dtype=float)
+        ys = np.linspace(-0.5, ny - 0.5, ninty, dtype=float)[1:-1]
+        nptx = xs.size
+        npty = ys.size
+
+        npts = 2 * (nptx + npty)
+
+        borderx = np.empty((npts + 1,), dtype=float)
+        bordery = np.empty((npts + 1,), dtype=float)
+
+        # "bottom" points:
+        borderx[:nptx] = xs
+        bordery[:nptx] = -0.5
```

Comment:
I suggest a comment explaining the -0.5 number or the use of a descriptive constant detailing what this number is.

Reply:
I agree, and I believe this has something to do with all the pixel offsetting. However this is largely just a copy of the stuff from JWST with minimal changes. If someone knows exactly what this is I would be happy to add a comment here.

Really, we should be maintaining a common "constants" object/module to document these and back reference that as opposed to having a bunch of ad hoc magic numbers some of which are documented with comments. This however, is something to consider later.

Reply:
Centers of pixels have integer image coordinates. Edges of the pixels are located `+/- 0.5` pixels away from centers. I think this convention is explained somewhere in the docs. IMO, it does not make sense to add a comment like this every single time we see a 0.5 in the code.

Reply:
and `-0.5` is shorter than `coordinate_of_left_pixel_border(_relative_to_center)`

Reply:
I agree with Mihai that in the context of pixels it's relatively clear what a 0.5 offset is doing. 
 However it looks to me like `calc_bounding_polygon` is the only function where this is really relevant, so would it work to add a note to the docstring saying something like,
"""
Notes
-----
The bounding polygon is defined from corners of pixels
whereas the reference pixel is defined at the center of a pixel
and therefore the lower-left corner is located at (-0.5, -0.5)
"""

or whatever you want, I'm sure I made some small inaccuracies in describing that but you get the idea

Reply:
"whereas the reference pixel" (in @emolter suggestion) should be something like "whereas the pixel coordinates refer to their centers" (or something like that). "Reference pixel" has a special meaning in the WCS world.

Reply:
I see that this was resolved but I don't see any docstring update, is there any objection to my previous suggestion?

Reply:
Oops I copied over that with something, I'll re-add this

Reply:
thanks!